tests/migration/paused.go

```diff
@@ -0,0 +1,190 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package migration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"kubevirt.io/kubevirt/tests/libmigration"
+
+	migrationsv1 "kubevirt.io/api/migrations/v1alpha1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+
+	kvpointer "kubevirt.io/kubevirt/pkg/pointer"
+
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/testsuite"
+
+	"kubevirt.io/kubevirt/tests/framework/checks"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"kubevirt.io/kubevirt/tests/libvmifact"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libnet"
+)
+
+var _ = SIGMigrationDescribe("Live Migrate A Paused VMI", func() {
+	var (
+		virtClient kubecli.KubevirtClient
+		err        error
+	)
+
+	BeforeEach(func() {
+		checks.SkipIfMigrationIsNotPossible()
+		virtClient = kubevirt.Client()
+	})
+
+	waitUntilMigrationMode := func(vmi *v1.VirtualMachineInstance, expectedMode v1.MigrationMode, timeout int) *v1.VirtualMachineInstance {
+		By("Waiting until migration status")
+		EventuallyWithOffset(2, func() v1.MigrationMode {
+			By("Retrieving the VMI post migration")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			if vmi.Status.MigrationState != nil {
+				return vmi.Status.MigrationState.Mode
+			}
+			return v1.MigrationPreCopy
+		}, timeout, 1*time.Second).Should(Equal(expectedMode), fmt.Sprintf("migration should be in %s after %d s", expectedMode, timeout))
+		return vmi
+	}
+
+	Describe("Starting a VirtualMachineInstance ", func() {
+
+		Context("paused vm during migration", func() {
+
+			var migrationPolicy *migrationsv1.MigrationPolicy
+
+			BeforeEach(func() {
+				By("enable AllowWorkloadDisruption and limit migration bandwidth")
+				policyName := fmt.Sprintf("testpolicy-%s", rand.String(5))
+				migrationPolicy = kubecli.NewMinimalMigrationPolicy(policyName)
+				migrationPolicy.Spec.AllowWorkloadDisruption = kvpointer.P(true)
+				migrationPolicy.Spec.AllowPostCopy = kvpointer.P(false)
+				migrationPolicy.Spec.CompletionTimeoutPerGiB = kvpointer.P(int64(20))
+				migrationPolicy.Spec.BandwidthPerMigration = kvpointer.P(resource.MustParse("5Mi"))
+			})
+
+			Context("should migrate paused", func() {
+
+				applyMigrationPolicy := func(vmi *v1.VirtualMachineInstance) {
+					AlignPolicyAndVmi(vmi, migrationPolicy)
```

Comment:
`AlignPolicyAndVmi()` modifies the VMI's labels in order for the migration policy to actually match it, as written in its documentation:
https://github.com/kubevirt/kubevirt/blob/d096f82c3382cfaeda2467deeb711e20a5c59b4a/tests/migration/migration_policy.go#L100

What you can do is update the VMI object to make sure these labels are updated:
```go
vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Update(context.Background(), vmi, metav1.UpdateOptions{})
```

Reply:
BTW I wonder how the tests passed this way

Reply:
The VMI hasn't been created at that stage; it just has been defined.
I simplified the test. You are right, there was no need for any strategy or a VM at all.